spec/models/post_spec.rb

```diff
@@ -419,6 +419,22 @@
       post = create(:post, content: '')
       expect(post.id).not_to be_nil
     end
+
+    it "should not allow adding a coauthor who's blocked you" do
+      author = create(:user)
+      coauthor = create(:user)
+      create(:block, blocking_user: coauthor, blocked_user: author, block_interactions: true)
+      post = build(:post, user: author, unjoined_authors: [coauthor])
+      expect(post).not_to be_valid
+    end
+
+    it "should not allow adding a coauthor you've blocked" do
+      author = create(:user)
+      coauthor = create(:user)
+      create(:block, blocking_user: author, blocked_user: coauthor, block_interactions: true)
+      post = build(:post, user: author, unjoined_authors: [coauthor])
+      expect(post).not_to be_valid
+    end
```

Comment:
Can we also test this prevents you adding coauthors who blocked / are blocked by some other coauthor in the post? That is, create a post, create a reply by a new author, create some block between this new author and a new user, then try amending the post to have the new user as an unjoined author.